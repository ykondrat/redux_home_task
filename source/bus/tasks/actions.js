// Types
import { types } from './types';

// Async
export const fetchTasksAsync = () => ({
    type: types.FETCH_TASKS_ASYNC
});
export const createTaskAsync = (message) => ({
    type: types.CREATE_TASK_ASYNC,
    payload: message
});
export const removeTaskAsync = (taskId) => ({
    type: types.REMOVE_TASK_ASYNC,
    payload: taskId
});
export const updateTaskAsync = (task) => ({
    type: types.UPDATE_TASK_ASYNC,
    payload: task
});
export const completeTasksAsync = (tasks) => ({
    type: types.COMPLETE_TASKS_ASYNC,
    payload: tasks
});

// Sync
export const fillTasks = (tasks) => ({
    type: types.FILL_TASKS,
    payload: tasks
});
export const createTask = (task) => ({
    type: types.CREATE_TASK,
    payload: task
});
export const removeTask = (taskId) => ({
    type: types.REMOVE_TASK,
    payload: taskId
});
export const updateTask = (task) => ({
    type: types.UPDATE_TASK,
    payload: task
});
export const completeTask = () => ({
    type: types.COMPLETE_TASKS,
});
