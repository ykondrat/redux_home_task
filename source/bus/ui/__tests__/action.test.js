import * as actions from "../actions";

describe("ui actions tests:", () => {
    test("startFetching test", () => {
        expect(actions.startFetching()).toMatchSnapshot();
    });
    test("stopFetching test", () => {
        expect(actions.stopFetching()).toMatchSnapshot();
    });
});
