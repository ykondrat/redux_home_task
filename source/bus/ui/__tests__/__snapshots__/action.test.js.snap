// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions tests: startFetching test 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`ui actions tests: stopFetching test 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
